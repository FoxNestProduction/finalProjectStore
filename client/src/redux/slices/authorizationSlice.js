import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isUserAuthorized: false,
  token: null,
};

const authorizationSlice = createSlice({
  name: 'authorization',
  initialState,
  reducers: {
    setAuthorization(state, action) {
      state.isUserAuthorized = action.payload;
    },
    setToken: (state, action) => {
      state.token = action.payload;
    },
  },

});

export const {
  setAuthorization,
  setToken,
} = authorizationSlice.actions;

export default authorizationSlice.reducer;
