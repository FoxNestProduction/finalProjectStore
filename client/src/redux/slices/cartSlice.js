import { createSlice } from '@reduxjs/toolkit';
import { useSelector } from 'react-redux';
import axios from 'axios';
import { allProducts } from './productsSlice';

const initialState = {
  cartItems: [],
  isLoading: false,
  isCart: false,
};

/* eslint-disable no-param-reassign */

const cartSlice = createSlice({
  name: 'cart',
  initialState,
  reducers: {
    addToCart(state, action) {
      // 1-ий варіант - коли нам в action.payload надходить повний об'єкт item
      if (state.cartItems.length === 0 && action.payload !== null) {
        state.cartItems.push(action.payload);
      } else {
        const index = state.cartItems.findIndex((item) => item.id === action.payload.id);
        if (index === -1) {
          state.cartItems.push(action.payload);
        } else {
          state.cartItems[index].quantity += 1;
        }
      }
      /** 2-ий варіант, коли нам приходить суто id товара потрібно звернутись
       * до однієї з двох написаних нижче функцій:
       * - deleteOrAddCartByItemId , або
       * - deleteOrAddFromCartByItemIdWithValueFromState
       * Потрібно буде протестувати, який варіант зручніше, той і використовувати
       */
    },
    setCart(state, action) {
      if (state.cartItems.length === 0) {
        state.cartItems = action.payload;
      } else {
        const uniqueFilteredProducts = action.payload.products.filter((product) => {
          const matchedProduct = state.cartItems
            .find((cartItem) => cartItem.item.id !== product.id);
          const mark = matchedProduct !== undefined;
          return mark;
        });
        const notUniqueFilteredProducts = action.payload.products.filter((product) => {
          const matchedProduct = state.cartItems
            .find((cartItem) => cartItem.item.id === product.id);
          const mark = matchedProduct !== undefined;
          return mark;
        }).map((product) => {
          const matchedProduct = state.cartItems
            .find((cartItem) => cartItem.item.id === product.id);
          product.quantity += matchedProduct.quantity;
          return product;
        });
        state.cartItems = { ...uniqueFilteredProducts, ...notUniqueFilteredProducts };
      }
    },
    setIsLoading(state, action) {
      state.isLoading = action.payload;
    },
    setIsCart(state, action) {
      state.isCart = action.payload;
    },
    deleteFromCart(state, action) {
      // 1 - ий варіант - коли нам в action.payload надходить повний об'єкт item
      if (state.cartItems.length) {
        const index = state.cartItems.findIndex((item) => item.id === action.payload.id);

        if (index !== -1) {
          state.cartItems[index] = (
            state.cartItems[index] === 1
              ? state.cartItems.splice(index, 1)
              : state.cartItems[index].quantity -= 1
          );
        }
      }
      /** 2-ий варіант, коли нам приходить суто id товара потрібно звернутись
         * до однієї з двох написаних нижче функцій:
         * - deleteOrAddCartByItemId , або
         * - deleteOrAddFromCartByItemIdWithValueFromState
         * Потрібно буде протестувати, який варіант зручніше, той і використовувати
         */
    },
  },
});

export const { addToCart, setCart, setIsLoading, deleteFromCart, setIsCart } = cartSlice.actions;

export const getCartItemsFromServer = () => async (dispatch) => {
  try {
    dispatch(setIsLoading(true));
    const userId = useSelector((state) => state.user.user.id);
    const { data } = await axios.get('http://localhost:4000/cart');

    if (data !== null && data !== undefined) {
      const cartContainer = data.find((cartObj) => cartObj.customerId.id === userId);

      if (cartContainer !== null && data !== undefined) {
        dispatch(setIsCart(true));
        dispatch(setCart(cartContainer));
      } else {
        dispatch(setIsCart(false));
      }
    } else {
      dispatch(setIsCart(false));
    }
    // якщо data undefined, то нічого не передаємо в кошик
    // а isCart(true or false). Код далі не виконується і дані на сторінці відмальовуються
    // тільки зі стора

    // dispatch(setCart(cartContainer)); // якщо ми тут оновимо state то оновиться і localStorage
    //  тому тут потрібно написати логіку злиття даних з localStorage з даними з сервера.
    dispatch(setIsLoading(false));
  } catch (error) {
    console.warn('Error loading cart:', error);
    dispatch(setIsLoading(false));
    dispatch(setIsCart(false));
  }
};

/** 2 варіанти функції якщо ми додаємо або видаляємо товар в state cart через прокидування id:
 * TODO: варіант A:
 * ми робимо підписку на необхідний нам state всіх продуктів в самій функції.
 * Знаходимо по id необхідний нам об'єкт, та прокидуємо його в один з reducer-ов
 * за допомогою одного з ключів DELETE або ADD. Необхідний нам об'єкт падає
 * в визначений reducer та додається до кошику, або ж видаляється з нього.
 * Ця функція передається через dispatch. В неї прокидується id та key.
 * dispatch(deleteOrAddCartByItemId(id, key))
 */
export const deleteOrAddCartByItemId = (id, key) => (dispatch) => {
  const products = useSelector((state) => state.products.products);
  if (products.length !== 0) {
    const cartItem = products.find((product) => product.id === id);
    if (cartItem !== null && cartItem !== undefined) {
      if (key === 'ADD') {
        dispatch(addToCart(cartItem));
      }
      if (key === 'DELETE') {
        dispatch(deleteFromCart(cartItem));
      }
    }
  }
};

/** 2 варіанти функції якщо ми додаємо або видаляємо товар в state cart через прокидування id:
 * TODO: варіант B:
 * всередині компонента товара ми робимо підписку через useSelector на цей самий об'єкт товара.
 * Якщо точніше, то це не зовсім підписка, а пошук через функціональний селектор необхідних нам
 * даних в останній версії store, завдяки функції useSelector. В самому компоненті потрібно буде
 * прописати такий код:
 * TODO: const fullItemObject = useSelector(deleteOrAddFromCartByItemIdWithValueFromState(id))
 * в константу fullItemObject прийде повний об'єкт цього товара, і при кліку на кнопку
 * "Додати до кошика" (в нашому випадку це знак +), або видалення з кошика
 * ми передаємо в onClick такий код:
 * додати до кошика
 * TODO: onClick={dispatch(addToCart(fullItemObject))}
 * або такий, якщо видалити з кошика
 * TODO: onClick={dispatch(deleteFromCart(fullItemObject))}
 * Попередньо імпортувавши dispatch та addToCart()/deleteFromCart()
 */

export const deleteOrAddFromCartByItemIdWithValueFromState = (id) => (state) => {
  const products = allProducts(state);
  console.log(products);
  if (products.length !== 0) {
    const cartItem = products.find((product) => product.id === id);
    console.log(cartItem);
    if (cartItem !== null && cartItem !== undefined) {
      return cartItem;
    }
    return null;
  }
  return null;
};

export default cartSlice.reducer;
