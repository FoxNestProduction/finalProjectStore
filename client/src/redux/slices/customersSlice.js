import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
  customers: [],
};

/* eslint-disable no-param-reassign */

const customersSlice = createSlice({
  name: 'customers',
  initialState,
  reducers: {
    setCustomers(state, action) { // eslint-disable-line no-shadow
      state.customers = action.payload;
    },
  },
});

export const getCustomers = () => async (dispatch) => {
  try {
    const { data } = await axios.get('http://localhost:4000/api/customers');
    console.log(data);
    dispatch(setCustomers(data));// eslint-disable-line no-use-before-define
  } catch (error) {
    console.log('%cError loading customers:', 'color: red; font-weight: bold;', error);
  }
};

export const { setCustomers } = customersSlice.actions;

/* eslint-enable no-param-reassign */

export default customersSlice.reducer;
