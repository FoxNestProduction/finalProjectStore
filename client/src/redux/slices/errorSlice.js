import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  error: '',
};

/* eslint-disable no-param-reassign */

const errorSlice = createSlice({
  name: 'error',
  initialState,
  reducers: {
    setError: (state, action) => {
      state.error = action.payload;
    },
  },

});

export const {
  setError,
} = errorSlice.actions;

/* eslint-enable no-param-reassign */

export default errorSlice.reducer;
