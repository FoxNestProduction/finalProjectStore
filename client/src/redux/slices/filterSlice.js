import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { instance } from '../../API/instance';
import { resetSearch, setInputSearchValue, setSearch } from './searchSlice';
import { setError, setLoading } from '../extraReducersHelpers';

export const fetchFilteredProducts = createAsyncThunk(
  'filter/fetchFilteredProducts',
  async (queryString, { rejectWithValue, dispatch }) => {
    try {
      const response = await instance.get(`/products/filter${queryString}`);
      if (response.data.products.length > 0) {
        // dispatch(setSearch([]));
        // dispatch(setInputSearchValue(''));
        dispatch(resetSearch());
      }
      console.log(response.data);
      return response.data;
    } catch (err) {
      return rejectWithValue(err.response.data);
    }
  },
);

const initialState = {
  filteredProducts: [],
  productsQuantity: null,
  loading: false,
  error: null,
  nothingFound: false,
  filterParams: {
    filterCategories: [],
    isTrending: false,
    rating: null, // mostPopular
    isHealthy: false,
    isSupreme: false,
    minPrice: 0,
    maxPrice: 30,
    sort: '',
    startPage: 1,
    perPage: 10,
  },
};

const filterSlice = createSlice({
  name: 'filter',
  initialState,
  reducers: {
    setFilteredProducts(state, action) {
      state.filteredProducts = action.payload;
    },
    setProductsQuantity(state, action) {
      state.productsQuantity = action.payload;
    },
    setNothingFound(state, action) {
      state.nothingFound = action.payload;
    },
    setFilterParams(state, action) {
      console.log('action.payload', action.payload);
      const newState = { ...state.filterParams, ...action.payload };
      console.log('newState', newState);
      state.filterParams = newState;
      // state.filterParams = { ...state.filterParams, ...action.payload };
    },
    resetFilter(state) {
      state.filteredProducts = [];
      state.productsQuantity = null;
      state.nothingFound = false;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchFilteredProducts.pending, setLoading)
      .addCase(fetchFilteredProducts.fulfilled, (state, action) => {
        if (action.payload.length === 0) {
          state.filteredProducts = [];
          state.nothingFound = true;
        } else {
          state.filteredProducts = action.payload.products;
          state.productsQuantity = action.payload.productsQuantity;
          state.nothingFound = false;
        }
        state.loading = false;
      })
      .addCase(fetchFilteredProducts.rejected, setError);
  },
});

export const { setFilteredProducts, setFilterParams,
  setNothingFound, setProductsQuantity, resetFilter } = filterSlice.actions;

export default filterSlice.reducer;
