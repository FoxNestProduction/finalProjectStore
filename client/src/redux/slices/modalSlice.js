import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isOpen: false,
  title: '',
  content: '',
  buttonBox: false,
  buttonAgree: {
    text: '',
    startIcon: false,
    endIcon: false,
    onClick: null,
  },
};

/* eslint-disable no-param-reassign */

const modalSlice = createSlice({
  name: 'modal',
  initialState,
  reducers: {
    openModal(state) {
      state.isOpen = true;
    },
    closeModal(state) {
      state.isOpen = false;
      state.title = '';
      state.content = '';
      state.buttonBox = false;
      state.buttonAgree.text = '';
      state.buttonAgree.startIcon = false;
      state.buttonAgree.endIcon = false;
      state.buttonAgree.onClick = null;
    },
    setTitle(state, action) {
      state.title = action.payload;
    },
    setContent(state, action) {
      state.content = action.payload;
    },
    setButtonAgree(state, action) {
      state.buttonAgree = action.payload;
    },
    addButtonBox(state) {
      state.buttonBox = true;
    },
  },
});

export const {
  openModal,
  closeModal,
  setTitle,
  setContent,
  setButtonAgree,
  addButtonBox,
} = modalSlice.actions;

/* eslint-enable no-param-reassign */

export default modalSlice.reducer;
