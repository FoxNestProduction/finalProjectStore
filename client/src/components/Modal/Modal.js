import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import IconButton from '@mui/material/IconButton';
import DialogContent from '@mui/material/DialogContent';
import DialogActions from '@mui/material/DialogActions';
import DialogContentText from '@mui/material/DialogContentText';
import Button from '@mui/material/Button';
import DeleteIcon from '@mui/icons-material/Delete';
import SendIcon from '@mui/icons-material/Send';
import CloseIcon from '@mui/icons-material/Close';

import {
  openModal,
  closeModal,
  setTitle,
  setContent,
  setButtonAgree,
  addButtonBox,
} from '../../redux/slices/modalSlice';
import { setAuthorizationError, setRegistrationError } from '../../redux/slices/errorSlice';

const Modal = () => {
  const dispatch = useDispatch();

  const isOpen = useSelector((state) => state.modal.isOpen);
  const title = useSelector((state) => state.modal.title);
  const content = useSelector((state) => state.modal.content);
  const buttonAgree = useSelector((state) => state.modal.buttonAgree);
  const buttonBox = useSelector((state) => state.modal.buttonBox);

  const handleClose = () => {
    dispatch(closeModal());
    dispatch(setAuthorizationError(''));
    dispatch(setRegistrationError(''));
  };

  return (
    <Dialog
      open={isOpen}
      onClose={handleClose}
    >
      <IconButton
        sx={{
          ml: 'auto',
          color: 'primary.main',
        }}
        onClick={handleClose}
      >
        <CloseIcon />
      </IconButton>
      {title && (
      <DialogTitle
        sx={{
          textAlign: 'center',
          bgcolor: 'primary.main',
          color: 'text.primaryLight',
          mb: 5,
        }}
      >
        {title}
      </DialogTitle>
      )}
      <DialogContent sx={{ textAlign: 'center', minHeight: '12vh' }}>
        {content}
      </DialogContent>
      {buttonBox && (
      <DialogActions>
        <Button
          sx={{ px: 1 }}
          variant="outlined"
          onClick={handleClose}
        >
          Close
        </Button>
        <Button
          sx={{ px: 1 }}
          variant="contained"
          endIcon={buttonAgree.endIcon && <SendIcon />}
          startIcon={buttonAgree.startIcon && <DeleteIcon />}
          onClick={buttonAgree.onClick}
          disabled={buttonAgree.disabled}
          autoFocus
        >
          {buttonAgree.text}
        </Button>
      </DialogActions>
      )}
    </Dialog>
  );
};

export default Modal;
