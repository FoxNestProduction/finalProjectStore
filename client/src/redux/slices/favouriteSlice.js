import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { useSelector } from 'react-redux';
import { setUser } from './userSlice';
import { setAuthorizationError } from './errorSlice';

const initialState = {
  favourites: [],
  cardStates: {},
};

const favouriteSlice = createSlice({
  name: 'favourites',
  initialState,
  reducers: {
    setFavourite(state, action) {
      state.favourites = action.payload;
      // localStorage.setItem('favourites', JSON.stringify(state.favourites));
      action.payload.forEach((id) => {
        state.cardStates[id] = true;
      });
    },
    addFavourite(state, action) {
      const { id } = action.payload;
      state.favourites.push(action.payload.id);
      // localStorage.setItem('favourites', JSON.stringify(state.favourites));
      state.cardStates[id] = true;
    },
    removeFavourite(state, action) {
    // eslint-disable-next-line no-underscore-dangle
      const { id } = action.payload;
      const newStateFavourites = state.favourites.filter((item) => item !== action.payload.id);
      state.favourites = newStateFavourites;
      // localStorage.setItem('favourites', JSON.stringify(newStateFavourites));
      state.cardStates[id] = false;
    },
    updateFavourites(state, action) {
      state.favourites = action.payload;
    },
  },
});

export const {
  setIsFavourite,
  addFavourite,
  removeFavourite,
  setFavourite,
  isFavourite,
  updateFavourites,
} = favouriteSlice.actions;

export default favouriteSlice.reducer;
