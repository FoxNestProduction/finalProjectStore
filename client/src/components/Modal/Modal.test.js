import React from 'react';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import Modal from './Modal';
import { closeModal } from '../../redux/slices/modalSlice';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));
jest.mock('../../redux/slices/modalSlice', () => ({
  ...jest.requireActual('../../redux/slices/modalSlice'),
  closeModal: jest.fn(),
}));
const mockDispatch = jest.spyOn(require('react-redux'), 'useDispatch');

describe('Modal Component', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should renders Modal component with provided data', async () => {
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Title');
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Content');
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce({});
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);   

    const component = render(
      <Modal />
    );

    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
    expect(component).toMatchSnapshot();
  });

  test('should handles button click', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Title');
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Content');
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce({});
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);   
  
      const component = render(
        <Modal />
      );
  
      jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);
     
      const modal = screen.getByTestId('CloseIcon');
      fireEvent.click(modal);

    await waitFor(() => {
      expect(dispatch).toHaveBeenCalledTimes(3);
      expect(closeModal).toHaveBeenCalled();
    });
  });

  test('should renders buttons and handles clicks', async () => {
    const dispatch = jest.fn();
    mockDispatch.mockReturnValueOnce(dispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Title');
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce('Content');
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce({});
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce({
      endIcon: true,
      startIcon: false,
      onClick: jest.fn(),
      disabled: false,
      text: 'Agree',
    });

    const component = render(<Modal />);
    expect(component).toMatchSnapshot();

    const closeButton = screen.getByText('Close');
    expect(closeButton).toBeInTheDocument();

    fireEvent.click(closeButton);
    await waitFor(() => {
    expect(closeModal).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledTimes(3);
    });

    const agreeButton = screen.getByRole('buttonAgree');
    expect(agreeButton).toBeInTheDocument();

    fireEvent.click(agreeButton);

    await waitFor(() => {
      expect(closeModal).toHaveBeenCalled();
      expect(dispatch).toHaveBeenCalledTimes(3);
    });
  })
});