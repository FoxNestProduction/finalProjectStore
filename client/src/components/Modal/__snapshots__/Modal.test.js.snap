// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should renders Modal component with provided data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-4h1gu1-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-dptng5-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            Title
          </h2>
          <div
            class="MuiDialogContent-root css-qvj5wr-MuiDialogContent-root"
          >
            Content
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal Component should renders buttons and handles clicks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r2:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-4h1gu1-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-dptng5-MuiTypography-root-MuiDialogTitle-root"
            id=":r2:"
          >
            Title
          </h2>
          <div
            class="MuiDialogContent-root css-qvj5wr-MuiDialogContent-root"
          >
            Content
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1gy6z57-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Close
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-focusVisible MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1b8eh7w-MuiButtonBase-root-MuiButton-root"
              role="buttonAgree"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              >
                <span
                  class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                  style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
                >
                  <span
                    class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
                  />
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
